test/unit/alphabet/detail/comparison_test.cpp

```diff
@@ -8,23 +8,78 @@
 #include <gtest/gtest.h>
 
 #include <seqan3/alphabet/detail/concept.hpp>
+#include <seqan3/test/concept_helper_classes.hpp>
 
-#include "../../std/concept/auxiliary.hpp"
-
+// Requires that two types can be compared for equality with each other (in either order) using both `==` and `!=`.
 TEST(comparison_concepts, weakly_equality_comparable_with)
 {
+    EXPECT_FALSE((seqan3::detail::weakly_equality_comparable_with<type_a, type_a>));
```

Comment:
`type_a` has:

```
    friend constexpr bool operator<(type_a const & lhs, type_a const & rhs);
    friend constexpr bool operator>(type_a const & lhs, type_a const & rhs);
    friend constexpr bool operator<=(type_a const & lhs, type_a const & rhs);
    friend constexpr bool operator>=(type_a const & lhs, type_a const & rhs);
```

Which then means that `seqan3::detail::weakly_ordered_with<type_a, type_a>` is true, because there are these 4 operators.

But `seqan3::detail::weakly_equality_comparable_with` is false, because `type_a == type_a` and `type_a != type_a` are not defined.

The other cases either have some declared operator to equality compare to `type_a` or can be compared via the base class, etc.